src/Robo/Hooks/CommandEventHook.php

```diff
@@ -41,4 +43,28 @@ public function issueWarnings(ConsoleCommandEvent $event) {
     $this->getInspector()->issueEnvironmentWarnings($command_name);
   }
 
+  /**
+   * Ask users to set a telemetry preference.
+   *
+   * @hook command-event *
+   */
+  public function telemetry(ConsoleCommandEvent $event) {
+    $userConfig = new UserConfig(Blt::configDir());
+
+    // Ask to enable telemetry if necessary.
+    if (!$userConfig->isTelemetrySet() && $this->input()->isInteractive()) {
+      $this->say("We strive to give you the best tools for development.\nYou can really help us improve by sharing anonymous performance and usage data.\n");
+      $preference = $this->confirm("Do you want to help us make this tool even better?", TRUE);
+      $userConfig->setTelemetryEnabled($preference);
+      if ($preference) {
+        $this->say("🎉 Awesome! Thank you for helping out!");
```

Comment:
Is there a command to opt out in case you're changing your mind?

Reply:
Good point, there is not. I'll create a follow up: https://github.com/acquia/blt/issues/3922